fixed the failing user tests - negative create, test_user_list

tkolhar opened the pull request:
please review it
thanks

elyezer commented on tests/foreman/cli/test_user.py:
```diff
@@ -646,7 +646,6 @@ def test_negative_create_user_3(self, opts):
         'email@brazil.b',
         '{0}@example.com'.format(
             gen_string('alpha', 49)),  # total length 61
-        '',
```
Empty email is now being accepted?

elyezer commented on tests/foreman/cli/test_user.py:
```diff
@@ -675,6 +674,36 @@ def test_negative_create_user_4(self, email):
+        @Feature: User - Negative Create
+
+        @Steps:
+        1. Create User for all invalid Email Address in [2]
```
This test is just using one data entry, then the `@data` decorator is not needed you can use the `''` string in the test body. Like:

```
options = {
    # ...
    'mail': '',
    # ...
}

elyezer commented:
I can't get the idea of this PR, will NACK until I get a better understanding of what is the idea. By lokking here I can assume that for any reason passing the `''` as the email address is not working but nothing more. Also the test which checks that failure needs to get its docstring updated.

elyezer commented on tests/foreman/cli/test_user.py:
```diff
@@ -675,6 +674,33 @@ def test_negative_create_user_4(self, email):
         self.assertTrue(result.stderr)
 
+    @skip_if_bug_open('bugzilla', 1204686)
+    def test_bugzilla_1204686(self, email):
```
This test will fail because was left the `email` argument.

tkolhar commented:
@elyezer : empty  email addresses should not be accepted but they are working hence
i filed the bz : 1204686 and added the test

Second scenario: email addresses with latin and utf-8 should be accepted but they are not 
working hence i filed the bz : 1204667 and added the test

Ichimonji10 commented on tests/foreman/cli/test_user.py:
```diff
@@ -675,6 +674,33 @@ def test_negative_create_user_4(self, email):
+        1. Create User with empty Email Address in [2]
+        using valid Username, First Name, Surname, Language, authorized by
+
+        @Assert: User is not created. Appropriate error shown.
```
How are you verifying that an appropriate error is shown? You can use `self.assertIn('desired error message', result.stdout)` to verify that some particular error message is shown.

Ichimonji10 commented on tests/foreman/cli/test_user.py:
```diff
@@ -675,6 +674,33 @@ def test_negative_create_user_4(self, email):
+        }
+        result = UserObj().create(options)
+        self.assertNotEqual(result.return_code, 0)
+        self.assertTrue(result.stderr)
```
Is `assertTrue` the best assertion to use here? How about `self.assertGreaterThan(len(result.stderr), 0)` or `self.assertIn('desired error message', result.stdout)`.

Ichimonji10 commented on tests/foreman/cli/test_user.py:
```diff
@@ -1657,6 +1681,42 @@ def test_list_user_4(self, test_data):
+        result = UserObj.list({
+            u'search': u'mail = {0}'.format(test_data['mail']),
+        })
+        self.assertEqual(len(result.stderr), 0)
```
How about adding `self.assertEqual(result.status_code, 0)`?

Ichimonji10 commented on tests/foreman/cli/test_user.py:
```diff
@@ -1657,6 +1681,42 @@ def test_list_user_4(self, test_data):
+            user = make_user(test_data)
+        except CLIFactoryError as err:
+            self.fail(err)
+        self.__assert_exists(user)
```
Why is a method that begins with a double underscore used?

tkolhar replied:
@Ichimonji10  there is already a method https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_user.py#L33 written i just used it

Ichimonji10 replied:
OK. That makes sense. Thank you.

Ichimonji10 commented:
ACK. Please rebase.

elyezer commented:
ACK
